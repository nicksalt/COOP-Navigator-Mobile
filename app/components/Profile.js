import React from "react";
import {
  View,
  ScrollView,
  Text,
  TouchableOpacity,
  ActivityIndicator,
  StyleSheet
} from "react-native";
import { Linking } from "expo";
import Hyperlink from "react-native-hyperlink";
import Field from "./Field";
import { PROFILE_URI } from "../utils/constants";

export default class Profile extends React.Component {
  downloadResume = () => {
    //this just opens the browser with the given url which downloads the file
    //to download the link directly (i.e. to download folder) we have to eject project
    Linking.openURL(PROFILE_URI + "/7680262/resume");
  };

  render() {
    const { isFetching, errorOccurred, studentProfile } = this.props;
    if (errorOccurred) {
      return (
        <View>
          <Text>There was an error getting your profile:</Text>
          <Text>{this.props.error.message}</Text>
        </View>
      );
    }
    if (isFetching || !studentProfile) {
      return (
        <View style={styles.activityIndicatorContainer}>
          <ActivityIndicator size="large" color="gray" />
        </View>
      );
    }

    const { address } = studentProfile;
    const { street, city, stateOrProvince, country, postalCode } =
      address || {};
    return (
      <ScrollView style={styles.profileContainer}>
        <Text style={styles.subtitle}>Personal Information</Text>
        <View style={styles.container}>
          <Field field="Student Number" fieldValue={studentProfile.studentId} />
          <Field
            field="Program of Studies"
            fieldValue={studentProfile.programOfStudy}
          />
        </View>
        <View style={styles.container}>
          <Field field="Title" fieldValue={studentProfile.title} />
          <Field field="First Name" fieldValue={studentProfile.firstName} />
          <Field field="Last Name" fieldValue={studentProfile.lastName} />
          <Field field="Gender" fieldValue={studentProfile.gender} />
        </View>
        <View style={styles.conatiner}>
          <Field field="Language" fieldValue={studentProfile.language} />
          <Field field="Citizenship" fieldValue={studentProfile.citizenship} />
        </View>
        <View style={styles.container}>
          <Field field="Accessibility" fieldValue="No" />
          <Field
            field="Resume"
            fieldValue={
              <TouchableOpacity onPress={this.downloadResume}>
                <Text style={styles.link}>Download</Text>
              </TouchableOpacity>
            }
          />
        </View>
        <Text style={styles.subtitle}>Contact Information</Text>
        <Field field="Email" fieldValue={studentProfile.email} />
        <Field
          field="LinkedIn"
          fieldValue={
            <Hyperlink linkDefault={true}>
              <Text style={styles.link}>{studentProfile.linkedIn}</Text>
            </Hyperlink>
          }
        />
        <Field field="Skype Username" fieldValue={studentProfile.skype} />
        <Field field="Phone Number" fieldValue={studentProfile.phoneNumber} />
        <Field
          field="Address"
          fieldValue={
            address &&
            `${street},
${city} ${stateOrProvince}, ${postalCode}`
          }
        />
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  profileContainer: {},
  subtitle: {
    padding: 10,
    fontSize: 16,
    fontWeight: "bold",
    textAlign: "center",
    color: "white",
    backgroundColor: "#a9343a"
  },
  container: {
    flexDirection: "row",
    justifyContent: "flex-start",
    flexWrap: "wrap"
  },
  switchContainer: {
    flexDirection: "row",
    justifyContent: "flex-start"
  },
  link: {
    color: "blue",
    textDecorationLine: "underline",
    fontSize: 18
  }
});
