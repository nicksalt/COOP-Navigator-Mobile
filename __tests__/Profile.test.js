import React from "react";
import Profile from "../app/components/Profile";
import renderer from "react-test-renderer";

jest.mock("ScrollView", () => "ScrollView");

test("Profile snapshot renders correctly", () => {
  const props = {};
  props.isFetching = true;
  let tree = renderer.create(<Profile {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.isFetching = false;
  props.errorOccurred = true;
  props.error = { message: "ERROR OCCURRED" };
  tree = renderer.create(<Profile {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.errorOccurred = false;
  props.studentProfile = {
    studentId: "7680262",
    title: "Mr.",
    lastName: "Mizon",
    firstName: "Alexandre",
    gender: "M",
    language: "English",
    citizenship: "Canadian",
    proofOfStatus: false,
    programOfStudy: "Software Engineering",
    requireAccessibleFacilities: false,
    placementIntention: false,
    email: "amizo067@uottawa.ca",
    password: "lethalpassword",
    linkedIn: "",
    skype: "",
    phoneNumber: "6142225555",
    address: {
      country: "Canada",
      stateOrProvince: "ON",
      city: "Ottawa",
      street: "800 King Edward Ave",
      postalCode: "K8H 4Z3"
    },
    termSequence: [
      {
        term: "2014, Fall",
        status: "study",
        programOfStudy: "Software Eng.",
        worktermReportGrade: "",
        course: ""
      },
      {
        term: "2015, Winter",
        status: "study",
        programOfStudy: "Software Eng.",
        worktermReportGrade: "",
        course: ""
      },
      {
        term: "2015, Summer",
        status: "work",
        programOfStudy: "Software Eng.",
        worktermReportGrade: "P",
        course: "SEG2901"
      }
    ]
  };
  tree = renderer.create(<Profile {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
