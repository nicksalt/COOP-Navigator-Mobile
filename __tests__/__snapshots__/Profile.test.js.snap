// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapshot renders correctly 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="gray"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Profile snapshot renders correctly 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    There was an error getting your profile:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    ERROR OCCURRED
  </Text>
</View>
`;

exports[`Profile snapshot renders correctly 3`] = `
<ScrollView
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Personal Information
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Student Number
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        7680262
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Program of Studies
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        Software Engineering
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Title
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        Mr.
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        First Name
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        Alexandre
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Last Name
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        Mizon
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Gender
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        M
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Language
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        English
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Citizenship
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        Canadian
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Accessibility
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        No
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 10,
          }
        }
      >
        Resume
      </Text>
      <View
        style={Object {}}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 18,
                "textDecorationLine": "underline",
              }
            }
          >
            Download
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Contact Information
  </Text>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      Email
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
        }
      }
    >
      amizo067@uottawa.ca
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      LinkedIn
    </Text>
    <View
      style={Object {}}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "blue",
              "fontSize": 18,
              "textDecorationLine": "underline",
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      Skype Username
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      Phone Number
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
        }
      }
    >
      6142225555
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      Address
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
        }
      }
    >
      800 King Edward Ave,
Ottawa ON, K8H 4Z3
    </Text>
  </View>
</ScrollView>
`;
